PiBridgeMaster.c

```diff
@@ -295,6 +295,19 @@ int PiBridgeMaster_Run(void)
 			if (bEntering_s) {
 				pr_info_master("Enter Init State\n");
 				bEntering_s = bFALSE;
+
+				/*
+				 * The Sniff2A pin has no external pull-down. It
+				 * can happen that the pin keeps its capacity
+				 * for a longer time after the pin has been
+				 * driven high. Thus we might detect a module
+				 * even if none is connected.
+				 * Configure a pull-down on the pin to drain any
+				 * capacity.
+				 */
+				if (!piDev_g.only_left_pibridge)
+					gpiod_set_config(piCore_g.gpio_sniff2a, PIN_CONFIG_BIAS_PULL_DOWN);
+
 				// configure PiBridge Sniff lines as input
```

Comment:
IMHO its cleaner if we set the initial state of the sniff pins unconditional

Reply:
OK, this was actually a bug. As I mixed right and left side up.
But it turned out, that on the Connect(+) there is a 10k pull-down on the SNIFF2A.
It is correct to not do this on a Connect(+). But it should also not hut doing it as a 100k pull-down would not be an issue. I will remove the condition.

Reply:
This feels like it doesn't belong in the state machine. I'd recommend moving this to `init_sniff_gpios()` inside `revpi_core.c`. Set it once on probe, not once on every bus scan.

Neither the code comment nor the commit message reveal which products are affected.

If this concerns only specific products (your comment above seems to indicate that the Connect(+) is not affected), by all means make it conditional on `if (piDev_g.machine_type == REVPI_CONNECT_4)` (or whatever other product needs it) to make it clear that you're working around a quirk of specific products. The if-clause also serves as documentation which products have the quirk. In general it seems like a bad idea to apply the quirk on unaffected products.

Reply:
Your right the `init_sniff_gpios()` should be the right place.

The only device which seems to have a pull-down on SNIFF2A is the Connect(+). And it looks like this was done more by accident then by choice.
As it should not be an issue to put an additional weak pull-down on the pin I thought of leaving any special treatment out.
On the other hand it is unclear why the combination of pull-ups and -downs was used on the pibridge.
Maybe we should figure out what the H/W effects are with the pulls. I will talk with the H/W guys about that. But this may take some time. Because of some vacations.